src/Thermal/PolytropicEOS.cpp

```diff
@@ -0,0 +1,117 @@
+//=================================================================================================
+//  PolytropicEOS.cpp
+//  Contains all function definitions for the Polytropic Equation of state.
+//
+//  This file is part of GANDALF :
+//  Graphical Astrophysics code for N-body Dynamics And Lagrangian Fluids
+//  https://github.com/gandalfcode/gandalf
+//  Contact : gandalfcode@gmail.com
+//
+//  Copyright (C) 2013  D. A. Hubber, G. Rosotti
+//
+//  GANDALF is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 2 of the License, or
+//  (at your option) any later version.
+//
+//  GANDALF is distributed in the hope that it will be useful, but
+//  WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//  General Public License (http://www.gnu.org/licenses) for more details.
+//=================================================================================================
+
+
+#include <math.h>
+#include "EOS.h"
+
+
+//=================================================================================================
+//  Polytropic::Polytropic()
+/// Default constructor for Polytropic EOS.  Passes and sets important
+/// thermal physics variables, as well as scaling to dimensionless units.
+//=================================================================================================
+template <int ndim>
+Polytropic<ndim>::Polytropic(FLOAT _Kpoly, FLOAT _eta, FLOAT _gamma, SimUnits *units):
+  EOS<ndim>(_gamma),
+  eta(_eta),
+  Kpoly(_Kpoly)
+{
+  //temp0 = temp0aux/units->temp.outscale;
+}
+
+
+
+//=================================================================================================
+//  Polytropic::Polytropic()
+/// Polytropic EOS destructor
+//=================================================================================================
+template <int ndim>
+Polytropic<ndim>::~Polytropic()
+{
+}
+
+
+
+//=================================================================================================
+//  Polytropic::Pressure
+/// Calculates and returns thermal pressure of referenced particle
+//=================================================================================================
+template <int ndim>
+FLOAT Polytropic<ndim>::Pressure(Particle<ndim> &part)
+{
+  return Kpoly*pow(part.rho, eta);
```

Comment:
This is the only EOS that does not define pressure as (gamma-1)rho*u. Even the isothermal EOS does that - the reason is that when we call pressure u and rho have already been computed. For this reason, in the neighbour manager branch I actually got rid of the pressure function from the children and only defined it in the parent class. I think the same will happen also to this one